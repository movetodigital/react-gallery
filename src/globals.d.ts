declare module '@storybook/addon-console';
