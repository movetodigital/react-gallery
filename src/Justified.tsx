import * as React from 'react';
import styled from 'styled-components';

type FunctionName = (n: object) => any;

interface ImageItemProps {
  src?: string;
  title?: string;
  alt?: string;
}

interface ComponentProps {
  images: Array<ImageItemProps>;
  gridColumns: number;
  gridGutter: number;
  aspectRatio?: number;
  imageRenderer?: FunctionName;
}

interface StyleItemProps {
  readonly width: number;
  readonly padding: string;
  readonly ratio: number;
}

interface RendererProps {
  props: ImageItemProps;
  key: number;
}

const Images = styled.div`
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
`;

const Item = styled.div<StyleItemProps>`
  color: white;
  position: relative;
  box-sizing: border-box;
  width: ${props => props.width}%;
  margin: ${props => props.padding};

  > div {
    width: 100%;
    padding-top: ${props => props.ratio}%;
  }
`;

const DefaultImage = styled.img`
  width: 100%;
  height: 100%;
  object-position: 50% 50%;
  object-fit: cover;
`;

const DefaultImageContainer = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
`;

const defaultRender = ({props: {src, title, alt}, key}: RendererProps) => {
  return (
    <DefaultImageContainer>
      <DefaultImage key={src} src={src} title={title} alt={alt} />
    </DefaultImageContainer>
  );
};

const Justified: React.FunctionComponent<ComponentProps> = ({
  images,
  gridColumns,
  gridGutter,
  aspectRatio = 1,
  imageRenderer = defaultRender,
}) => {
  const gutter = gridGutter / 2;
  const width = (100 - (gridColumns - 1) * gutter * 2) / gridColumns;
  const ratio = 100 / aspectRatio;

  return (
    <Images>
      {images.map((el, index) => (
        <Item
          key={`${index}_${gutter}`}
          width={width}
          padding={`${gutter}% ${
            (index + 1) % gridColumns === 0 ? '0' : gutter
          }% ${gutter}% ${index % gridColumns === 0 ? '0' : gutter}%`}
          ratio={ratio}
        >
          <div>{imageRenderer({props: el, key: index})}</div>
        </Item>
      ))}
    </Images>
  );
};

export {Justified};
