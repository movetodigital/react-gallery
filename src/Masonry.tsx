import * as React from 'react';
import styled from 'styled-components';

type FunctionName = (n: object) => any;

interface ImageItemProps {
  src?: string;
  title?: string;
  alt?: string;
}

interface ComponentProps {
  images: Array<ImageItemProps>;
  gridColumns: number;
  gridGutter: number;
  imageRenderer?: FunctionName;
}

interface ImagesComponentProps {
  gridGutter: number;
  gridColumns: number;
}

interface StyleItemProps {
  gridGutter: number;
}

const Images = styled.div<ImagesComponentProps>`
  column-count: ${props => props.gridColumns};
  margin: 0 auto;
  column-gap: ${props => props.gridGutter}%;
`;

const Item = styled.div<StyleItemProps>`
  margin: 0 0 ${props => props.gridGutter}em;
`;

const DefaultImage = styled.img`
  max-width: 100%;
  max-height: 100%;
`;

interface RendererProps {
  props: ImageItemProps;
  key: number;
}

const defaultRender = ({props: {src, title, alt}, key}: RendererProps) => {
  return <DefaultImage key={src} src={src} title={title} alt={alt} />;
};

const Masonry: React.FunctionComponent<ComponentProps> = ({
  images,
  gridColumns,
  gridGutter,
  imageRenderer = defaultRender,
}) => {
  return (
    <Images gridColumns={gridColumns} gridGutter={gridGutter}>
      {images.map((el, index) => (
        <Item key={`${index}_${gridGutter}`} gridGutter={gridGutter}>
          <div>{imageRenderer({props: el, key: index})}</div>
        </Item>
      ))}
    </Images>
  );
};

export {Masonry};
