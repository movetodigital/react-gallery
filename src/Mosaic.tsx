import * as React from 'react';
import styled from 'styled-components';

type FunctionName = (n: object) => any;

interface ImageItemProps {
  src: string;
  title?: string;
  alt?: string;
}

interface ComponentProps {
  images: Array<ImageItemProps>;
  gridColumns: number;
  gridGutter: number;
  imageRenderer?: FunctionName;
}

const Images = styled.div`
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
`;

interface StyleItemProps {
  readonly width: number;
  readonly padding: string;
  readonly height: number;
}

const Item = styled.div<StyleItemProps>`
  color: white;
  position: relative;
  box-sizing: border-box;
  width: ${props => props.width}%;
  margin: ${props => props.padding};

  > div {
    height: ${props => props.height}px;
  }
`;

const DefaultImage = styled.img`
  width: 100%;
  height: 100%;
  object-fit: cover !important;
  overflow: hidden !important;
`;

interface RendererProps {
  props: ImageItemProps;
  key: number;
}

const defaultRender = ({props: {src, title, alt}, key}: RendererProps) => {
  return <DefaultImage key={src} src={src} title={title} alt={alt} />;
};

const Mosaic: React.FunctionComponent<ComponentProps> = ({
  images,
  gridColumns,
  gridGutter,
  imageRenderer = defaultRender,
}) => {
  const gutter = gridGutter / 2;
  const width = (100 - (gridColumns - 1) * gutter * 2) / gridColumns;
  const height = 1200 / gridColumns;

  return (
    <Images>
      {images.map((el, index) => (
        <Item
          key={`${index}_${gutter}`}
          width={width}
          padding={`${gutter}% ${
            (index + 1) % gridColumns === 0 ? '0' : gutter
          }% ${gutter}% ${index % gridColumns === 0 ? '0' : gutter}%`}
          height={height}
        >
          <div>{imageRenderer({props: el, key: index})}</div>
        </Item>
      ))}
    </Images>
  );
};

export {Mosaic};
