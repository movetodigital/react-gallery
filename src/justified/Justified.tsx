/* eslint-disable @typescript-eslint/no-explicit-any */
// @ts-nocheck
import React, {useCallback, useEffect, useRef, useState} from 'react';

import {Container} from '../common/nodes';
import {defaultDetailsViewRenderer, defaultRenderer} from '../common/Renderers';
import ViewMonitor from '../common/ViewMonitor';
import Engine, {
  COLUMN_MAX_HEIGHT,
  COLUMN_MAX_WIDTH,
  COLUMNS_MAX_COUNT,
  GUTTER_IN_PERCENT,
  PLACEHOLDER_COLOR,
  VIEWPORT_HEIGHT,
  VIEWPORT_WIDTH,
} from './Engine';

type FunctionName = (n: object) => any;

interface ImageItemProps {
  src?: string;
  title?: string;
  alt?: string;
}

interface ComponentProps {
  imageRenderer?: FunctionName;
  images?: Array<ImageItemProps>;
  columnsMaxCount?: number;
  columnMaxWidth?: number;
  columnMaxHeight?: number;
  gutterInPercent?: number;
  className?: string;
  columnClassName?: string;
  rowClassName?: string;
  disableObserver?: boolean;
  disableActualImage?: boolean;
  fixedBottom?: number;
  enableDetailView?: boolean;
  detailsViewRenderer?: FunctionName;
  disableLastRowDetecting?: boolean;
  placeholderColor?: string;
  viewportWidth?: number;
  viewportHeight?: number;
  fixedBottomGutterInPercent?: number;
  fixedImagePlaceholderColor?: string;
}

const Justified = ({
  imageRenderer = defaultRenderer,
  images = [],
  columnsMaxCount = COLUMNS_MAX_COUNT,
  columnMaxWidth = COLUMN_MAX_WIDTH,
  columnMaxHeight = COLUMN_MAX_HEIGHT,
  gutterInPercent = GUTTER_IN_PERCENT,
  className = '',
  columnClassName = '',
  rowClassName = '',
  disableObserver = false,
  disableActualImage = false,
  enableDetailView = false,
  detailsViewRenderer = defaultDetailsViewRenderer,
  disableLastRowDetecting = false,
  placeholderColor = PLACEHOLDER_COLOR,
  viewportWidth = VIEWPORT_WIDTH,
  viewportHeight = VIEWPORT_HEIGHT,
}: ComponentProps) => {
  const engine = useRef<any>();
  const container: React.MutableRefObject<
    HTMLDivElement | undefined
  > = useRef();
  const wheelRef = useRef<any>();

  const [selectedImageRow, setSelectedImageRow] = useState(null);
  const [selectedImage, setSelectedImage] = useState(null);
  const [selectedRowHeight, setSelectedRowHeight] = useState(0);
  const [selectedImageId, setSelectedImageId] = useState(null);
  const [selectedImageProps, setSelectedImageProps] = useState({});
  const [rows, setRows] = useState([]);
  const [scrollTop, setScrollTop] = useState(0);
  const [containerWidth, setContainerWidth] = useState(0);

  const resize = () => {
    if (container.current) {
      setContainerWidth(container.current.offsetWidth);
    }
  };

  const onWheel = useCallback(
    e => {
      const amount = e.deltaY ? e.deltaY : e.deltaX;
      const newScrollTop = Math.max(scrollTop + amount, scrollTop);
      setScrollTop(newScrollTop);
    },
    [scrollTop, setScrollTop]
  );

  useEffect(() => {
    wheelRef.current = onWheel;
  }, [onWheel]);

  useEffect(() => {
    // Create event listener that calls handler function stored in ref
    const eventListener = (event: any) => wheelRef.current(event);
    if (container.current) {
      container.current.addEventListener('wheel', eventListener, {
        passive: false,
      });
    }

    return () => {
      if (container.current) {
        container.current.removeEventListener('wheel', eventListener);
      }
    };
  }, [container]);

  useEffect(() => {
    window.addEventListener('resize', resize);
    resize();
    engine.current = new Engine();
    engine.current
      .setImages(images)
      .setMaxColumnsCount(columnsMaxCount)
      .setColumnMaxWidth(columnMaxWidth)
      .setColumnMaxHeight(columnMaxHeight)
      .setGutterInPercent(gutterInPercent)
      .setViewportWidth(viewportWidth)
      .setScrollTop(scrollTop)
      .setViewportHeight(viewportHeight)
      .setDisableLastRowDetecting(disableLastRowDetecting)
      .setParentWidth((container.current || {}).offsetWidth);

    setRows(engine.current.buildFastRows());

    return () => window.removeEventListener('resize', resize);
  }, []);

  useEffect(() => {
    engine.current
      .setImages(images)
      .setMaxColumnsCount(columnsMaxCount)
      .setColumnMaxWidth(columnMaxWidth)
      .setColumnMaxHeight(columnMaxHeight)
      .setGutterInPercent(gutterInPercent)
      .setViewportWidth(viewportWidth)
      .setScrollTop(scrollTop)
      .setViewportHeight(viewportHeight)
      .setDisableLastRowDetecting(disableLastRowDetecting)
      .setParentWidth(containerWidth);
    setRows(engine.current.buildFastRows());
  }, [
    images,
    columnsMaxCount,
    columnMaxWidth,
    columnMaxHeight,
    gutterInPercent,
    viewportWidth,
    scrollTop,
    containerWidth,
    viewportHeight,
    disableLastRowDetecting,
  ]);

  const handleSelectImage = ({
    imageRow,
    imageId,
    rowHeight,
    image,
    imageProps,
  }: {
    imageRow: any;
    imageId: any;
    rowHeight: any;
    image: any;
    imageProps: any;
  }) => {
    if (selectedImageRow !== imageRow || imageId !== selectedImageId) {
      setSelectedImageRow(imageRow);
      setSelectedImage(image);
      setSelectedRowHeight(rowHeight);
      setSelectedImageId(imageId);
      setSelectedImageProps(imageProps);
    } else {
      setSelectedImageRow(null);
      setSelectedImage(null);
      setSelectedRowHeight(0);
      setSelectedImageId(null);
      setSelectedImageProps({});
    }
  };

  return (
    <Container
      // @ts-ignore
      ref={container}
      className={className}
      viewportHeight={viewportHeight}
      onWheel={onWheel}
    >
      {rows.map((el: any, rowIndex: number) => {
        const row = el.row;
        // @ts-ignore
        return (
          /* eslint-disable-next-line react/no-array-index-key */
          <React.Fragment key={`row-${rowIndex}`}>
            <div className={rowClassName}>
              {row.map((column: any, columnIndex: number) => {
                const newWidth = engine.current.calculateWidth(
                  column,
                  row,
                  el.isIncomplete
                );
                const newHeight = engine.current.calculateHeight(
                  column,
                  row,
                  el.isIncomplete
                );
                const newWidthInPercent = engine.current.getNewWidthInPercent(
                  newWidth,
                  el.isIncomplete,
                  row
                );
                const placeholderHeight = (100 * newHeight) / newWidth;

                return (
                  <div
                    /* eslint-disable-next-line react/no-array-index-key */
                    key={`column-${column.src}-${rowIndex}-${columnIndex}`}
                    className={columnClassName}
                    // @ts-ignore
                    style={{
                      verticalAlign: 'top',
                      position: 'relative',
                      display: 'inline-block',
                      width:
                        el.isIncomplete && !el.isIncomplete
                          ? `${newWidth}px`
                          : `${newWidthInPercent}%`,
                      maxWidth:
                        el.isIncomplete && !el.isIncomplete
                          ? `${newWidthInPercent}%`
                          : null,
                      margin:
                        row.length === columnIndex + 1
                          ? `0 0 ${engine.current.getGutterInPercent()}% 0`
                          : `0 ${engine.current.getGutterInPercent()}% ${engine.current.getGutterInPercent()}% 0`,
                    }}
                  >
                    <ViewMonitor
                      disableObserver={disableObserver}
                      disableActualImage={disableActualImage}
                    >
                      {(isViewable: any) =>
                        imageRenderer({
                          ...column,
                          newWidth,
                          newHeight,
                          newWidthInPercent,
                          placeholderHeight,
                          placeholderColor,
                          visible: !disableActualImage && isViewable,
                          enableMasonry: false,
                          onClick: () =>
                            handleSelectImage({
                              imageRow: rowIndex,
                              image: column,
                              rowHeight: newHeight,
                              // eslint-disable-next-line
                              imageId: `column-${column.src}-${rowIndex}-${columnIndex}`,
                              imageProps: {
                                placeholderHeight,
                                newWidthInPercent,
                                newWidth,
                                newHeight,
                              },
                            }),
                        }) as React.ReactElement
                      }
                    </ViewMonitor>
                  </div>
                );
              })}
            </div>
            <div key={`detail-${rowIndex}`}>
              {enableDetailView &&
                detailsViewRenderer({
                  ...row,
                  visible: rowIndex === selectedImageRow,
                  selectedImage,
                  rowHeight: selectedRowHeight,
                  gutterInPercent: engine.current.getGutterInPercent(),
                  selectedImageId,
                  selectedImageProps,
                })}
            </div>
          </React.Fragment>
        );
      })}
    </Container>
  );
};

export {Justified};
