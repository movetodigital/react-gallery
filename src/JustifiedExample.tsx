import * as React from 'react';

interface Props {
  margin?: number;
}

import {Justified} from './Justified';

const getRandomInt = (max: number) => {
  return Math.floor(Math.random() * Math.floor(max));
};

const images = Array.from('x'.repeat(40)).map(item => ({
  src: `https://source.unsplash.com/800x${800 + getRandomInt(100)}`,
  title: 'hello',
  alt: 'hello',
}));
export const JustifiedExample: React.FunctionComponent<Props> = ({margin}) => (
  <div style={{width: '1200px', margin}}>
    <Justified
      aspectRatio={16 / 9}
      gridColumns={5}
      gridGutter={1}
      images={images}
    />
  </div>
);
