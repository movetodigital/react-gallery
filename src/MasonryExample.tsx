import * as React from 'react';

interface Props {
  margin?: number;
}

import {Masonry} from './Masonry';

const getRandomInt = (max: number) => {
  return Math.floor(Math.random() * Math.floor(max));
};

const images = Array.from('x'.repeat(40)).map(item => ({
  src: `https://source.unsplash.com/800x${800 + getRandomInt(700)}`,
  title: 'hello',
  alt: 'hello',
}));
export const MasonryExample: React.FunctionComponent<Props> = ({margin}) => (
  <div style={{width: '1200px', margin}}>
    <Masonry gridColumns={5} gridGutter={1} images={images} />
  </div>
);
