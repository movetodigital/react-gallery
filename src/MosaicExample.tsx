import * as React from 'react';

interface Props {
  margin?: number;
}

import {Mosaic} from './Mosaic';

const images = Array.from('x'.repeat(40));
export const MosaicExample: React.FunctionComponent<Props> = ({margin}) => (
  <div style={{maxWidth: '400px', margin}}>
    <Mosaic images={images} />
  </div>
);
