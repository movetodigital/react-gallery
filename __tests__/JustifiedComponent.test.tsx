import {shallow} from 'enzyme';
import * as React from 'react';

import {Justified} from '../src/Justified';

const hydratedProps = {
  images: [
    {
      src: '/image/name/1',
      width: 500,
      height: 500,
    },
    {
      src: '/image/name/1',
      width: 500,
      height: 500,
    },
    {
      src: '/image/name/1',
      width: 500,
      height: 500,
    },
    {
      src: '/image/name/1',
      width: 500,
      height: 500,
    },
  ],
  gridColumns: 5,
  gridGutter: 1,
};

describe('<Justified />', () => {
  it('should render and match snapshot', () => {
    const tree = shallow(<Justified {...hydratedProps} />);

    expect(tree).toMatchSnapshot();
  });
});
