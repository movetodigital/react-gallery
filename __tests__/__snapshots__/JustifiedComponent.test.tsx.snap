// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Justified /> should render and match snapshot 1`] = `
<Justified
  gridColumns={5}
  gridGutter={1}
  images={
    Array [
      Object {
        "height": 500,
        "src": "/image/name/1",
        "width": 500,
      },
      Object {
        "height": 500,
        "src": "/image/name/1",
        "width": 500,
      },
      Object {
        "height": 500,
        "src": "/image/name/1",
        "width": 500,
      },
      Object {
        "height": 500,
        "src": "/image/name/1",
        "width": 500,
      },
    ]
  }
>
  <Styled(Component)
    className=""
    onWheel={[Function]}
  >
    <StyledComponent
      className=""
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-htpNat",
            "isStatic": false,
            "lastClassName": "cihfGY",
            "rules": Array [
              "
  position: relative;
  display: block;
  font-size: 0;
  opacity: ",
              [Function],
              ";
  height: initial;
  overflow: initial;
",
            ],
          },
          "displayName": "Styled(Component)",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-htpNat",
          "target": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={
        Object {
          "current": <div
            class="sc-htpNat cihfGY"
          >
            <div
              class=""
            >
              <div
                class=""
                style="vertical-align: top; position: relative; display: inline-block; width: 20%; margin: 0px 0.5% 0.5% 0px;"
              >
                <div>
                  <img
                    alt=""
                    class="sc-bwzfXH eYmfNB"
                    src="/image/name/1"
                  />
                  <div
                    style="padding-top: 100%; background-color: rgb(240, 240, 240);"
                  />
                </div>
              </div>
              <div
                class=""
                style="vertical-align: top; position: relative; display: inline-block; width: 20%; margin: 0px 0.5% 0.5% 0px;"
              >
                <div>
                  <img
                    alt=""
                    class="sc-bwzfXH eYmfNB"
                    src="/image/name/1"
                  />
                  <div
                    style="padding-top: 100%; background-color: rgb(240, 240, 240);"
                  />
                </div>
              </div>
              <div
                class=""
                style="vertical-align: top; position: relative; display: inline-block; width: 20%; margin: 0px 0.5% 0.5% 0px;"
              >
                <div>
                  <img
                    alt=""
                    class="sc-bwzfXH eYmfNB"
                    src="/image/name/1"
                  />
                  <div
                    style="padding-top: 100%; background-color: rgb(240, 240, 240);"
                  />
                </div>
              </div>
              <div
                class=""
                style="vertical-align: top; position: relative; display: inline-block; width: 20%; margin: 0px 0px 0.5% 0px;"
              >
                <div>
                  <img
                    alt=""
                    class="sc-bwzfXH eYmfNB"
                    src="/image/name/1"
                  />
                  <div
                    style="padding-top: 100%; background-color: rgb(240, 240, 240);"
                  />
                </div>
              </div>
            </div>
            <div />
          </div>,
        }
      }
      onWheel={[Function]}
    >
      <ForwardRef
        className="sc-htpNat cihfGY"
        onWheel={[Function]}
      >
        <div
          className="sc-htpNat cihfGY"
        >
          <div
            className=""
          >
            <div
              className=""
              key="column-/image/name/1-0-0"
              style={
                Object {
                  "display": "inline-block",
                  "margin": "0 0.5% 0.5% 0",
                  "maxWidth": null,
                  "position": "relative",
                  "verticalAlign": "top",
                  "width": "20%",
                }
              }
            >
              <ViewMonitor>
                <div>
                  <Image
                    alt=""
                    aspectRatio={1}
                    enableMasonry={false}
                    height={200}
                    newHeight={200}
                    newWidth={200}
                    newWidthInPercent={20}
                    onClick={[Function]}
                    originalHeight={500}
                    originalWidth={500}
                    placeholderColor="#f0f0f0"
                    placeholderHeight={100}
                    src="/image/name/1"
                    visible={true}
                    width={200}
                  >
                    <Styled(Component)
                      alt=""
                      onClick={[Function]}
                      src="/image/name/1"
                      visible={true}
                    >
                      <StyledComponent
                        alt=""
                        forwardedComponent={
                          Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "attrs": Array [],
                            "componentStyle": ComponentStyle {
                              "componentId": "sc-bwzfXH",
                              "isStatic": false,
                              "lastClassName": "eYmfNB",
                              "rules": Array [
                                "
  position: absolute;
  display: ",
                                [Function],
                                ";
  width: 100%;
  cursor: pointer;
",
                              ],
                            },
                            "displayName": "Styled(Component)",
                            "foldedComponentIds": Array [],
                            "render": [Function],
                            "styledComponentId": "sc-bwzfXH",
                            "target": [Function],
                            "toString": [Function],
                            "warnTooManyClasses": [Function],
                            "withComponent": [Function],
                          }
                        }
                        forwardedRef={null}
                        onClick={[Function]}
                        src="/image/name/1"
                        visible={true}
                      >
                        <Component
                          alt=""
                          className="sc-bwzfXH eYmfNB"
                          onClick={[Function]}
                          src="/image/name/1"
                          visible={true}
                        >
                          <img
                            alt=""
                            className="sc-bwzfXH eYmfNB"
                            onClick={[Function]}
                            src="/image/name/1"
                          />
                        </Component>
                      </StyledComponent>
                    </Styled(Component)>
                  </Image>
                  <div
                    style={
                      Object {
                        "backgroundColor": "#f0f0f0",
                        "paddingTop": "100%",
                      }
                    }
                  />
                </div>
              </ViewMonitor>
            </div>
            <div
              className=""
              key="column-/image/name/1-0-1"
              style={
                Object {
                  "display": "inline-block",
                  "margin": "0 0.5% 0.5% 0",
                  "maxWidth": null,
                  "position": "relative",
                  "verticalAlign": "top",
                  "width": "20%",
                }
              }
            >
              <ViewMonitor>
                <div>
                  <Image
                    alt=""
                    aspectRatio={1}
                    enableMasonry={false}
                    height={200}
                    newHeight={200}
                    newWidth={200}
                    newWidthInPercent={20}
                    onClick={[Function]}
                    originalHeight={500}
                    originalWidth={500}
                    placeholderColor="#f0f0f0"
                    placeholderHeight={100}
                    src="/image/name/1"
                    visible={true}
                    width={200}
                  >
                    <Styled(Component)
                      alt=""
                      onClick={[Function]}
                      src="/image/name/1"
                      visible={true}
                    >
                      <StyledComponent
                        alt=""
                        forwardedComponent={
                          Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "attrs": Array [],
                            "componentStyle": ComponentStyle {
                              "componentId": "sc-bwzfXH",
                              "isStatic": false,
                              "lastClassName": "eYmfNB",
                              "rules": Array [
                                "
  position: absolute;
  display: ",
                                [Function],
                                ";
  width: 100%;
  cursor: pointer;
",
                              ],
                            },
                            "displayName": "Styled(Component)",
                            "foldedComponentIds": Array [],
                            "render": [Function],
                            "styledComponentId": "sc-bwzfXH",
                            "target": [Function],
                            "toString": [Function],
                            "warnTooManyClasses": [Function],
                            "withComponent": [Function],
                          }
                        }
                        forwardedRef={null}
                        onClick={[Function]}
                        src="/image/name/1"
                        visible={true}
                      >
                        <Component
                          alt=""
                          className="sc-bwzfXH eYmfNB"
                          onClick={[Function]}
                          src="/image/name/1"
                          visible={true}
                        >
                          <img
                            alt=""
                            className="sc-bwzfXH eYmfNB"
                            onClick={[Function]}
                            src="/image/name/1"
                          />
                        </Component>
                      </StyledComponent>
                    </Styled(Component)>
                  </Image>
                  <div
                    style={
                      Object {
                        "backgroundColor": "#f0f0f0",
                        "paddingTop": "100%",
                      }
                    }
                  />
                </div>
              </ViewMonitor>
            </div>
            <div
              className=""
              key="column-/image/name/1-0-2"
              style={
                Object {
                  "display": "inline-block",
                  "margin": "0 0.5% 0.5% 0",
                  "maxWidth": null,
                  "position": "relative",
                  "verticalAlign": "top",
                  "width": "20%",
                }
              }
            >
              <ViewMonitor>
                <div>
                  <Image
                    alt=""
                    aspectRatio={1}
                    enableMasonry={false}
                    height={200}
                    newHeight={200}
                    newWidth={200}
                    newWidthInPercent={20}
                    onClick={[Function]}
                    originalHeight={500}
                    originalWidth={500}
                    placeholderColor="#f0f0f0"
                    placeholderHeight={100}
                    src="/image/name/1"
                    visible={true}
                    width={200}
                  >
                    <Styled(Component)
                      alt=""
                      onClick={[Function]}
                      src="/image/name/1"
                      visible={true}
                    >
                      <StyledComponent
                        alt=""
                        forwardedComponent={
                          Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "attrs": Array [],
                            "componentStyle": ComponentStyle {
                              "componentId": "sc-bwzfXH",
                              "isStatic": false,
                              "lastClassName": "eYmfNB",
                              "rules": Array [
                                "
  position: absolute;
  display: ",
                                [Function],
                                ";
  width: 100%;
  cursor: pointer;
",
                              ],
                            },
                            "displayName": "Styled(Component)",
                            "foldedComponentIds": Array [],
                            "render": [Function],
                            "styledComponentId": "sc-bwzfXH",
                            "target": [Function],
                            "toString": [Function],
                            "warnTooManyClasses": [Function],
                            "withComponent": [Function],
                          }
                        }
                        forwardedRef={null}
                        onClick={[Function]}
                        src="/image/name/1"
                        visible={true}
                      >
                        <Component
                          alt=""
                          className="sc-bwzfXH eYmfNB"
                          onClick={[Function]}
                          src="/image/name/1"
                          visible={true}
                        >
                          <img
                            alt=""
                            className="sc-bwzfXH eYmfNB"
                            onClick={[Function]}
                            src="/image/name/1"
                          />
                        </Component>
                      </StyledComponent>
                    </Styled(Component)>
                  </Image>
                  <div
                    style={
                      Object {
                        "backgroundColor": "#f0f0f0",
                        "paddingTop": "100%",
                      }
                    }
                  />
                </div>
              </ViewMonitor>
            </div>
            <div
              className=""
              key="column-/image/name/1-0-3"
              style={
                Object {
                  "display": "inline-block",
                  "margin": "0 0 0.5% 0",
                  "maxWidth": null,
                  "position": "relative",
                  "verticalAlign": "top",
                  "width": "20%",
                }
              }
            >
              <ViewMonitor>
                <div>
                  <Image
                    alt=""
                    aspectRatio={1}
                    enableMasonry={false}
                    height={200}
                    newHeight={200}
                    newWidth={200}
                    newWidthInPercent={20}
                    onClick={[Function]}
                    originalHeight={500}
                    originalWidth={500}
                    placeholderColor="#f0f0f0"
                    placeholderHeight={100}
                    src="/image/name/1"
                    visible={true}
                    width={200}
                  >
                    <Styled(Component)
                      alt=""
                      onClick={[Function]}
                      src="/image/name/1"
                      visible={true}
                    >
                      <StyledComponent
                        alt=""
                        forwardedComponent={
                          Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "attrs": Array [],
                            "componentStyle": ComponentStyle {
                              "componentId": "sc-bwzfXH",
                              "isStatic": false,
                              "lastClassName": "eYmfNB",
                              "rules": Array [
                                "
  position: absolute;
  display: ",
                                [Function],
                                ";
  width: 100%;
  cursor: pointer;
",
                              ],
                            },
                            "displayName": "Styled(Component)",
                            "foldedComponentIds": Array [],
                            "render": [Function],
                            "styledComponentId": "sc-bwzfXH",
                            "target": [Function],
                            "toString": [Function],
                            "warnTooManyClasses": [Function],
                            "withComponent": [Function],
                          }
                        }
                        forwardedRef={null}
                        onClick={[Function]}
                        src="/image/name/1"
                        visible={true}
                      >
                        <Component
                          alt=""
                          className="sc-bwzfXH eYmfNB"
                          onClick={[Function]}
                          src="/image/name/1"
                          visible={true}
                        >
                          <img
                            alt=""
                            className="sc-bwzfXH eYmfNB"
                            onClick={[Function]}
                            src="/image/name/1"
                          />
                        </Component>
                      </StyledComponent>
                    </Styled(Component)>
                  </Image>
                  <div
                    style={
                      Object {
                        "backgroundColor": "#f0f0f0",
                        "paddingTop": "100%",
                      }
                    }
                  />
                </div>
              </ViewMonitor>
            </div>
          </div>
          <div
            key="detail-0"
          />
        </div>
      </ForwardRef>
    </StyledComponent>
  </Styled(Component)>
</Justified>
`;
